storage/innobase/handler/ha_innodb.cc

```diff
@@ -7019,7 +7019,7 @@ ha_innobase::open(
 
 	innobase_copy_frm_flags_from_table_share(ib_table, table->s);
 
-	if (ib_table->is_readable()) {
+	if (ib_table->is_readable() || dict_table_is_discarded(ib_table)) {
```

Comment:
Thanks for finding the root cause commit. That commit still gives sleepless nights.. (many coding guide violations) and also caused other bugs.

It looks like upstream calls this dict_stats_init() unconditionally. Can you make it unconditional and run all tests (specially the encryption tests)?

Reply:
As I see it if we can call stats init without IBD file, it should be fine with an unreadable file too?